src/analysis/special_functions/pow.lean

```diff
@@ -1598,6 +1598,40 @@ lemma filter.tendsto.ennrpow_const {α : Type*} {f : filter α} {m : α → ℝ
   tendsto (λ x, (m x) ^ r) f (𝓝 (a ^ r)) :=
 (ennreal.continuous_rpow_const.tendsto a).comp hm
 
+-- the following lemma is placed here instead of `analysis.specific_limits` since
+-- we need `real.rpow`
+/-- The sum of `2⁻¹ ^ i` for `n ≤ i` equals `2⁻¹ ^ (n - 1)`. -/
+lemma tsum_geometric_inv_two_ge (n : ℕ) :
+  ∑' i, ite (n ≤ i) ((2 : ℝ)⁻¹ ^ i) 0 = 2⁻¹ ^ (n - 1 : ℝ) :=
```

Comment:
The statement feels strange, because of this real power on the right hand side. Would it be enough for your application to say that the result is `2 * (2⁻¹ ^ n)` (which can be stated without real powers)? Or with `ℤ` powers?

Also, I think the proof can be simplified. What about starting for instance with
```lean
begin
  have A : summable (λ (i : ℕ), ite (n ≤ i) ((2⁻¹ : ℝ) ^ i) 0),
  { apply summable_of_nonneg_of_le _ _ summable_geometric_two;
    { assume i, by_cases hi : n ≤ i; simp [hi] } },
  have B : (finset.range n).sum (λ (i : ℕ), ite (n ≤ i) ((2⁻¹ : ℝ)^i) 0) = 0,
  { apply finset.sum_eq_zero,
    assume i hi,
    simp only [inv_eq_zero, inv_pow₀, ite_eq_right_iff],
    assume h,
    exact (lt_irrefl _ ((finset.mem_range.1 hi).trans_le h)).elim },
  rw [← sum_add_tsum_nat_add n A, B],
  simp only [if_true, zero_add, zero_le', le_add_iff_nonneg_left, pow_add, tsum_mul_right,
    tsum_geometric_inv_two],
  sorry,
```
(This can probably be simplified further).

Reply:
Thanks! This is a much better formulation